### PR TITLE
Nav: Fix selection state

### LINKED ISSUE #8137: Nav selection is broken when using url fragment identifier
### Environment Information 
 
- __Package version(s)__: 5.134.0

### Please provide a reproduction of the bug in a codepen:
 
https://codepen.io/anon/pen/MxwmeW
 
#### Actual behavior:
 
Selection is broken when still in the same page but using url property in NavLink with fragment identifiers. When clicking on a NavLink, then on another one, the previous is still selected.
 
#### Expected behavior:
 
Selection must be functional when clicking on the different NavLink items (according to me 🙂).

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: (Low) 
 
Products/sites affected: (if applicable)

### PR DESCRIPTION
**Pull request by leddie24**
#### Pull request checklist

- [x] Addresses an existing issue: Fixes #8137
- [x] Include a change request file using `$ npm run change`

#### Description of changes

Only check URL/location.has if state.selectedKey is undefined.

#### Focus areas to test

(optional)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/OfficeDev/office-ui-fabric-react/pull/8250)

### CODE REVIEW

**Comment by KevinTCoughlin** [comment]
If this approach is sound, I recommend adding a test as this seems like behavior that can be easily regressed.

**Comment by leddie24** [comment]
I'll wait for a code review from @jdhuntington before I commit to adding a test.

File: packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx [hunk 1]
```diff
@@ -316,20 +316,23 @@ export class NavBase extends BaseComponent<INavProps, INavState> implements INav
...
 
-    if (location.protocol + '//' + location.host + location.pathname === target) {
-      return true;
-    }
-
-    if (location.hash) {
-      // Match the hash to the url.
-      if (location.hash === link.url) {
+    // If selectedKey is not defined in state, then check URL to determine link selected status
+    if (!this.state.selectedKey) {
```
Comment (jdhuntington) [inline packages/office-ui-fabric-react/src/components/Nav/Nav.base.tsx]: Line 297 of this file (same method) is already checking `state.selectedKey`. Would be nice to move this logic up there and always return early if either `props.selectedKey` or `state.selectedKey` is set.

**Comment by leddie24** [comment]
Still adding tests! (sorry, this is taking me a while, still learning). I added a test to have the component check for location.href if state/props selectedKey isn't set.  I will add a test to make sure state.selectedKey takes precedence over location.href next.

**Comment by KevinTCoughlin** [comment]
> Still adding tests! (sorry, this is taking me a while, still learning). I added a test to have the component check for location.href if state/props selectedKey isn't set. I will add a test to make sure state.selectedKey takes precedence over location.href next.

Reach out if you need help writing the test, I've encountered most gotchyas at this point (I think). Don't want you to be blocked on this.

**Comment by leddie24** [comment]
Should be ready to re-review @KevinTCoughlin / @jdhuntington .  Let me know if there needs to be more tests or modifications.  Thanks!

**Pull request merged**